test/crdsvalidation/kongPluginBinding_test.go

```diff
@@ -0,0 +1,652 @@
+								Name:  "test-route",
+								Kind:  "KongRoute",
+								Group: "configuration.konghq.com",
+							},
+						},
+					},
+				},
+				expectedErrorMessage: lo.ToPtr(" KongRoute can be used only when serviceRef is unset or set to KongService"),
+			},
+			{
+				name: "configuration.konghq.com/KongService, networking.k8s.io/Ingress targets",
+				kongPluginBinding: configurationv1alpha1.KongPluginBinding{
+					ObjectMeta: commonObjectMeta,
+					Spec: configurationv1alpha1.KongPluginBindingSpec{
+						PluginReference: &configurationv1alpha1.PluginRef{
+							Kind: lo.ToPtr("KongPlugin"),
+							Name: "my-plugin",
+						},
+						Targets: &configurationv1alpha1.KongPluginBindingTargets{
+							ServiceReference: &configurationv1alpha1.TargetRefWithGroupKind{
+								Name:  "test-service",
+								Kind:  "KongService",
+								Group: "configuration.konghq.com",
+							},
+							RouteReference: &configurationv1alpha1.TargetRefWithGroupKind{
+								Name:  "test-route",
+								Kind:  "Ingress",
+								Group: "networking.k8s.io",
+							},
+						},
+					},
+				},
+				expectedErrorMessage: lo.ToPtr("KongService can be used only when routeRef is unset or set to KongRoute"),
+			},
+		},
+	}
+
+	wrongTargetsGroupKindTCs = testCasesGroup{
+		name: "targets group/kind validation",
+		testCases: []kongPluginBindingTC{
+			{
+				name: "networking.k8s.io/Ingress, as service target",
+				kongPluginBinding: configurationv1alpha1.KongPluginBinding{
+					ObjectMeta: commonObjectMeta,
+					Spec: configurationv1alpha1.KongPluginBindingSpec{
+						PluginReference: &configurationv1alpha1.PluginRef{
+							Kind: lo.ToPtr("KongPlugin"),
+							Name: "my-plugin",
+						},
+						Targets: &configurationv1alpha1.KongPluginBindingTargets{
+							ServiceReference: &configurationv1alpha1.TargetRefWithGroupKind{
+								Name:  "test-service",
+								Kind:  "Ingress",
+								Group: "networking.k8s.io",
+							},
+						},
+					},
+				},
+				expectedErrorMessage: lo.ToPtr("group/kind not allowed for the serviceRef"),
+			},
+			{
+				name: "core/Service, as route target",
+				kongPluginBinding: configurationv1alpha1.KongPluginBinding{
+					ObjectMeta: commonObjectMeta,
+					Spec: configurationv1alpha1.KongPluginBindingSpec{
+						PluginReference: &configurationv1alpha1.PluginRef{
+							Kind: lo.ToPtr("KongPlugin"),
+							Name: "my-plugin",
+						},
+						Targets: &configurationv1alpha1.KongPluginBindingTargets{
+							RouteReference: &configurationv1alpha1.TargetRefWithGroupKind{
+								Name:  "test-route",
+								Kind:  "Service",
+								Group: "core",
+							},
+						},
+					},
+				},
+				expectedErrorMessage: lo.ToPtr("group/kind not allowed for the routeRef"),
+			},
+		},
+	}
+)
+
+func init() {
+	kongPluginBindingsTestCases = append(kongPluginBindingsTestCases,
+		pluginRefTCs,
+		globalTargetTCs,
+		targetsCombinationsTCs,
+		crossTargetsTCs,
+		wrongTargetsGroupKindTCs,
+	)
+}
```

Comment:
Why do we need the `init` here?

Can't we just use the testcase variables inside the test? 🤔 

Alternative idea: instead of putting all those testcases in this file and making it that longer, perhaps we could add the `init` and a testcase each in a separate file. Then it's easier to read and easier to sport that the testcase needs to be added via `init`. WDYT?

Reply:
It's a good idea. I introduced a new package with all the test cases. LTKWYT